src/root/views/FinalReportForm/index.tsx

```diff
@@ -0,0 +1,566 @@
+import scrollToTop from '#utils/scrollToTop';
+
+import {
+  DrefFinalReportFields,
+  eventFields,
+  needsFields,
+  overviewFields,
+  operationFields,
+  submissionFields,
+  DrefFinalReportApiFields,
+  ONSET_IMMINENT,
+} from './common';
+import Overview from './Overview';
+import EventDetails from './EventDetails';
+import Needs from './Needs';
+import Operation from './Operation';
+import Submission from './Submission';
+import useDrefFinalReportFormOptions, { schema } from './useDreFinalReportOptions';
+
+import styles from './styles.module.scss';
+
+//function scrollToTop() {
+//  window.setTimeout(() => {
+//    window.scrollTo({
+//      top: Math.min(145, window.scrollY),
+//      left: 0,
+//      behavior: 'smooth',
+//    });
+//  }, 0);
+//}
+interface Props {
+  match: match<{ id?: string }>;
+  history: History;
+  location: Location;
+}
+interface DrefFinalResponseFields {
+  id: string;
+}
+
+type StepTypes = 'operationOverview' | 'eventDetails' | 'needs' | 'operation' | 'submission';
+
+const stepTypesToFieldsMap: {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  [key in StepTypes]: (keyof DrefFinalReportFields)[];
+} = {
+  operationOverview: overviewFields,
+  eventDetails: eventFields,
+  needs: needsFields,
+  operation: operationFields,
+  submission: submissionFields,
+};
+
+const defaultFormValues: PartialForm<DrefFinalReportFields> = {
+  photos: []
+};
+
+function FinalReport(props: Props) {
+  const {
+    match,
+  } = props;
+  const { id } = match.params;
+  const alert = useAlert();
+  const {
+    value,
+    error,
+    setFieldValue: onValueChange,
+    validate,
+    setValue,
+    setError,
+  } = useForm(
+    schema,
+    { value: defaultFormValues }
+  );
+
+  const {
+    countryOptions,
+    disasterCategoryOptions,
+    disasterTypeOptions,
+    fetchingCountries,
+    fetchingDisasterTypes,
+    fetchingDrefOptions,
+    fetchingUserDetails,
+    interventionOptions,
+    nationalSocietyOptions,
+    needOptions,
+    onsetOptions,
+    yesNoOptions,
+    userDetails,
+
+  } = useDrefFinalReportFormOptions(value);
+
+  const [fileIdToUrlMap, setFileIdToUrlMap] = useState<Record<number, string>>({});
+  const { strings } = useContext(languageContext);
+  const [currentStep, setCurrentStep] = useState<StepTypes>('operationOverview');
+  const submitButtonLabel = currentStep === 'submission' ? strings.drefFormSaveButtonLabel : strings.drefFormContinueButtonLabel;
+  const shouldDisabledBackButton = currentStep === 'operationOverview';
+
+  const erroredTabs = useMemo(() => {
+    const safeErrors = getErrorObject(error) ?? {};
+
+    const tabs: {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      [key in StepTypes]: boolean;
+    } = {
+      operationOverview: false,
+      eventDetails: false,
+      needs: false,
+      operation: false,
+      submission: false,
+    };
+
+    return mapToMap(
+      tabs,
+      (key) => key,
+      (_, tabKey) => {
```

Comment:
It's weird that the actual value is not used here. We can just use listToMap on keys of tabs instead.

Reply:
@frozenhelium